### DISCUSSION TITLE
Security issue with Casks

### DISCUSSION THREAD

**Discussion by quackerex** [Homebrew/discussions#5378] [Casks]
### Output of `brew config`

```shell
N/A
```

### Output of `brew doctor`

```shell
N/A
```

### Description of issue

It seems like giving "App management" or "Full Disk Access" to Terminal [is not secure](https://sector7.computest.nl/post/2024-04-bringing-process-injection-into-view-exploiting-all-macos-apps-using-nib-files/).

> Note that Homebrew [asks you](https://docs.brew.sh/FAQ#why-do-my-cask-apps-lose-their-dock-position--launchpad-position--permission-settings-when-i-run-brew-upgrade) to grant “App Management” (or “Full Disk Access”) permission to your terminal. This is a bad idea, as it would make you vulnerable to these attacks again: any non-sandboxed application can execute code with the TCC permissions of your terminal by adding a malicious command to (e.g.) ~/.zshrc. Granting “App Management” or “Full Disk Access” to your terminal should be considered the same as disabling TCC completely.

### Relevant casks

N/A

**Comment by p-linnane** [comment]
This is inaccurate. The FAQ states that it's an option to provide those permissions if you want in-place upgrades, but it's not required. Users are expected to understand the consequences of granting permissions to applications.

**Reply by quackerex** [reply]
I haven't tried Casks but if Homebrew doesn't asks for the permission I don't have a big issue.

> Users are expected to understand the consequences of granting permissions to applications.

Might be important to highlight the risk. I wasn't even aware of this before finding the article.
The main issue is not the permission itself but giving the permission to Terminal App. Which would allow any app in the system to gain the same privilege.
